### PR TITLE
Fix Template Inserter

### LINKED ISSUE #2887: E2E Tests: Template Insertion
In #2879 we added tests to verify that templates are being shown.

We should also verify that clicking on templates inserts them as skeletons.

### LINKED ISSUE #3016: Template Inserter broken
Steps to reproduce:
- Make sure Gutenberg is up to date.
- Open the Template Inserter

Screenshot 2019-08-12 at 21 59 56

Todo:

- [x] Fix Template Inserter
- [x] E2E tests (#2887)

### PR DESCRIPTION
**Pull request by miina**
Fixes #3016.
Closes #2887 

Todo:

- [x] Fix Template style.
- [x] E2E

### CODE REVIEW

File: tests/e2e/specs/stories-editor/story-templates.test.js [hunk 1]
```diff
@@ -61,5 +61,75 @@ describe( 'Story Templates', () => {
...
+
+			await openTemplateInserter();
+
+			// Wait until the templates are loaded and blocks accessible.
+			await page.waitForSelector( '.block-editor-block-preview .wp-block' );
+
+			const templateContents = await page.evaluate( () => {
+				const contents = [];
+				const previews = document.querySelectorAll( '.block-editor-block-preview .block-editor-inner-blocks.has-overlay' );
+				previews.forEach( function( preview, index ) {
```
Comment (swissspidy) [inline tests/e2e/specs/stories-editor/story-templates.test.js]: Sounds like an ideal use case for `Array.prototype.reduce` :-)

**Comment by swissspidy** [comment]
We might need to add another block deprecation for the quote block:

```
Content generated by `save` function:

<blockquote class="wp-block-quote has-text-align-center is-style-white"><p>Everyone soon or late comes round Rome</p><cite>Robert Browning</cite></blockquote>

Content retrieved from post body:

<blockquote style="text-align:center" class="wp-block-quote is-style-white"><p>Everyone soon or late comes round Rome</p><cite>Robert Browning</cite></blockquote>
```

Gutenberg is going away from these text-align inline styles.

**Comment by miina** [comment]
> Gutenberg is going away from these text-align inline styles.

If that happens then perhaps the deprecation will be implemented upstream instead?

**Comment by swissspidy** [comment]
They did: https://github.com/WordPress/gutenberg/pull/16779

Does that mean it's not working? 🤔  Or are we perhaps overriding the core deprecation?

**Comment by swissspidy** [comment]
In any case, we can look at that independent from this PR if needed.

**Comment by miina** [comment]
Will check if that's an easy fix.

**Comment by miina** [comment]
> Does that mean it's not working? 🤔 Or are we perhaps overriding the core deprecation?

It means that it was working properly, otherwise, it would have given a block validation error inside the template :)

Updated the template and seems to work OK now.

**Comment by kienstra** [comment]
**Approved**

Hi @miina,
This looks good. The Template Inserter now displays as expected, when using the lastest from the [master branch](https://github.com/WordPress/gutenberg/tree/master) of Gutenberg.

# Before

before-inserter-broken


# After 

template-inserter-works-expected

**Pull request merged**